backend/src/signer/signer.rs

```diff
@@ -0,0 +1,122 @@
+use crate::contracts::generated::summa_contract::Summa;
+use ethers::{
+    abi::{encode, Token},
+    prelude::SignerMiddleware,
+    providers::{Http, Middleware, Provider},
+    signers::{LocalWallet, Signer, WalletError},
+    types::{Address, Signature},
+    utils::keccak256,
+};
+use futures::future::join_all;
+use serde_json::Value;
+use std::{error::Error, fs::File, io::BufReader, path::Path, str::FromStr, sync::Arc};
+
+#[derive(Debug)]
+pub struct SummaSigner {
+    signing_wallets: Vec<LocalWallet>,
+    summa_contract: Summa<SignerMiddleware<Provider<Http>, LocalWallet>>,
+}
+
+impl SummaSigner {
+    pub fn new(
+        private_keys: &[&str],
+        main_signer_key: &str,
+        chain_id: u64,
+        rpc_url: &str,
+        address: Address,
+    ) -> Self {
+        let wallet: LocalWallet = LocalWallet::from_str(main_signer_key).unwrap();
+
+        let provider = Provider::<Http>::try_from(rpc_url).unwrap();
+        let client = Arc::new(SignerMiddleware::new(
+            provider,
+            wallet.with_chain_id(chain_id),
+        ));
+
+        let contract = Summa::new(address, client);
+        Self {
+            signing_wallets: private_keys
+                .iter()
+                .map(|private_key| LocalWallet::from_str(private_key).unwrap())
+                .collect(),
+            summa_contract: contract,
+        }
+    }
+
+    pub fn get_deployment_address<P: AsRef<Path>>(
```

Comment:
What is the role of this function? I don't see it used anywhere

Reply:
It will be used in backend to get the deployed Summa address from the json file.